### PR TITLE
Update to GWT 2.10.0

### PR DESCRIPTION
**Pull request by gtritchie**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/11505

### Approach

In our gwt fork repo, created `rstudio/ocean-storm` branch from the `rstudio/v1.5` branch, then merged the upstream gwt `2.10.0-google` tag into it.

Then tweaked our scripts and built GWT.

There was one unit test failure in GWT after this upgrade which I have fixed.

A StringIndexOutOfBoundsException was being thrown. In this build of GWT String.substring throws an exception if you pass it a start index greater than the end index. Previously it would return an empty string.

The "correct" modern behaviour is to throw an exception, but we might want to create a wrapper that implements the old behaviour to avoid a bunch of edge cases failing.

All the gory details are here:

<details>
<summary>Failed Test Stack Dump</summary>

```
     [java] 1) test17(org.rstudio.core.client.ConsoleOutputWriterTests)
     [java] java.lang.StringIndexOutOfBoundsException: fromIndex: 10, toIndex: 6, length: 23
     [java]     at java.lang.Throwable.Throwable(Throwable.java:68)
     [java]     at java.lang.Exception.Exception(Exception.java:29)
     [java]     at java.lang.RuntimeException.RuntimeException(RuntimeException.java:29)
     [java]     at java.lang.IndexOutOfBoundsException.IndexOutOfBoundsException(IndexOutOfBoundsException.java:29)
     [java]     at java.lang.StringIndexOutOfBoundsException.StringIndexOutOfBoundsException(StringIndexOutOfBoundsException.java:30)
     [java]     at javaemul.internal.InternalPreconditions.checkCriticalStringBounds(InternalPreconditions.java:585)
     [java]     at org.rstudio.core.client.VirtualConsole.$submit(InternalPreconditions.java:568)
     [java]     at org.rstudio.core.client.ConsoleOutputWriter.$outputToConsole(ConsoleOutputWriter.java:128)
     [java]     at org.rstudio.core.client.ConsoleOutputWriterTests.test17(ConsoleOutputWriterTests.java:659)
     [java]     at Unknown.anonymous(GWTTestMetadataImpl.java:107)
     [java]     at com.google.gwt.junit.client.impl.GWTTestAccessor.$invoke(GWTTestAccessor.java:35)
     [java]     at com.google.gwt.junit.client.impl.GWTRunner.$executeTestMethod(GWTRunner.java:226)
     [java]     at com.google.gwt.junit.client.GWTTestCase.$doRunTest(GWTTestCase.java:157)
     [java]     at com.google.gwt.junit.client.GWTTestCase.$runBare(TestCase.java:59)
     [java]     at com.google.gwt.junit.client.GWTTestCase.$__doRunTest(GWTTestCase.java:115)
     [java]     at com.google.gwt.junit.client.impl.GWTRunner.$runTest(GWTRunner.java:302)
     [java]     at com.google.gwt.junit.client.impl.GWTRunner.$doRunTest(GWTRunner.java:235)
     [java]     at com.google.gwt.junit.client.impl.GWTRunner$TestBlockListener.$onSuccess(GWTRunner.java:106)
     [java]     at com.google.gwt.junit.client.impl.GWTRunner$TestBlockListener.onSuccess(GWTRunner.java:100)
     [java]     at com.google.gwt.user.client.rpc.impl.RequestCallbackAdapter.onResponseReceived(RequestCallbackAdapter.java:232)
     [java]     at com.google.gwt.http.client.Request.$fireOnResponseReceived(Request.java:227)
     [java]     at com.google.gwt.http.client.RequestBuilder$1.onReadyStateChange(RequestBuilder.java:412)
     [java]     at Unknown.anonymous(XMLHttpRequest.java:329)
     [java]     at com.google.gwt.core.client.impl.Impl.apply(Impl.java:299)
     [java]     at com.google.gwt.core.client.impl.Impl.entry0(Impl.java:358)
     [java]     at Unknown.anonymous(Impl.java:78)
```

</details>

### Automated Tests

Existing GWT unit tests are all passing.

### QA Notes

Nothing to specifically test other than general usage. There is always a possibility that the stricter string checking that found the broken unit test could cause some user-facing features to have issues. These would likely turn out to be previously hidden bugs, so probably a Good Thing.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/gwt/test/org/rstudio/core/client/ConsoleOutputWriterTests.java [hunk 1]
```diff
@@ -656,18 +656,16 @@ public void test17()
       ConsoleOutputWriter output = getCOW();
       output.outputToConsole("✔ xxx \033[34myyy\033[39m xxx", myClass, notError, ignoreLineCount, false);
       output.outputToConsole("\r", myClass, notError, ignoreLineCount, false);
-      output.outputToConsole("✔xxx \033[31myyy\033[39m zzz", myClass, notError, ignoreLineCount, false);
+      output.outputToConsole("✔xxx \033[31myyy\033[39m zzz", myClass, notError, ignoreLineCount, false);
```
[the remark this answers is not in the document]
Comment (jgutman) [reply src/gwt/test/org/rstudio/core/client/ConsoleOutputWriterTests.java:659]: I'm not getting how this fixed the broken unit test

[the remark this answers is not in the document]
Comment (jgutman) [reply src/gwt/test/org/rstudio/core/client/ConsoleOutputWriterTests.java:659]: I can see that this did fix the failing unit test, and I can see in my IDE it looks like some kind of ESC code was removed, but I don't quite get how that relates to the string indexing and where in https://github.com/rstudio/rstudio/blob/5ecd209673e476549955cc5c8b2f636269c61712/src/gwt/src/org/rstudio/core/client/ConsoleOutputWriter.java#L90 we were relying on that old behavior

[the remark this answers is not in the document]
Comment (gtritchie) [reply src/gwt/test/org/rstudio/core/client/ConsoleOutputWriterTests.java:659]: If we were to proceed I would back out this unit test change (the actual fix would be to replace use of the String.substring with a new method that behaves the way it did prior to gwt 2.10).

But yes there were embedded literal escape characters (weren't intended to be there but went unnoticed for years). Their presence just happened to trigger a situation that was sensitive to the change in behaviour of String.substring, which was handy!

**Comment by kevinushey** [comment]
LGTM; does this also include updates for GIN / Guice?

**Comment by gtritchie** [comment]
Not proceeding with this for Ocean Storm. See the issue for some notes.
